python.md

```diff
@@ -974,6 +974,49 @@ from path.to.some.module import \
     thing1, thing2, thing3, thing4
 ```
 
+### Make function signatures explicit
+
+Avoid using ``**kwargs`` in function signatures without good reason. Code with loosely defined
+function signatures can be difficult to work with, as it's unclear what variables are
+entering the function.
+ 
+```python
+def do_something(**kwargs):  # Don't do this
+   ...
+```
+
+Be explicit instead:
+
+```python
+def do_something(foo: int, bar: str):
+   ...
+```
+
+This includes functions that wrap lower level functionality, such as model creation methods:
+
+```python
+class MyModel(models.Model):
+    ...
+    
+    @classmethod
+    def new(cls, **kwargs):  # Don't do this.
+        return cls.objects.create(**kwargs)
```

Comment:
Out of curiosity, what's the reason for not using `kwargs` in this case? Is it that it obfuscates the parameters that `new()` accepts, because you have to check the definition of `create()` to find out? Does it also defeat type checks?

Reply:
Yeah, it hurts any static analysis. Type checks for one, but also IDE tooltips and autocomplete